File: 14600/BOJ_14651.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    private static final int MOD = 1_000_000_009;
    private void solution() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        if (n==1) {
            System.out.println(0);
            return;
        }
        int answer = 2;
        for (int i = 2; i < n; i++) {
            answer = (int) (1l*answer*3%MOD);
        }
        System.out.println(answer);
    }

    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}

File: 14600/BOJ_14651.kt
import java.io.BufferedReader
import java.io.InputStreamReader

const val MOD = 1_000_000_009
fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    var dp = arrayOf(0, 1, 1)
    repeat(n-1) {
        var tmp = Array(3) {0}
        for (i in 0..2) {
            for (j in 0 .. 2) {
                val idx = (i+j)%3
                tmp[idx]+=dp[j]
                tmp[idx]%=MOD
            }
        }
        dp = tmp
    }
    println(dp[0])
}
